core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1599,9 +1616,17 @@ public function deleteChilds($table, $id, &$delete)
 
 			if ($objDelete->numRows && !($GLOBALS['TL_DCA'][$v]['config']['doNotDeleteRecords'] ?? null) && \strlen($v))
 			{
-				foreach ($objDelete->fetchAllAssoc() as $row)
+				$rows = $objDelete->fetchAllAssoc();
+				$this->preloadActiveRecords(array_map(static function ($row) { return $row['id']; }, $rows));
```

Comment:
could use `array_column` but it still fetches the records twice and we don't really need the `activeRecord` here?